### PR TITLE
core: api to pass additional information with standalone file URI

### PR DESCRIPTION
**Pull request by epoberezkin**

### CODE REVIEW

File: src/Simplex/Chat.hs [hunk 1]
```diff
@@ -3271,10 +3272,10 @@ processAgentMsgSndFile _corrId aFileId msg =
             Nothing -> do
               withAgent (`xftpDeleteSndFileInternal` aFileId)
               withStore' $ \db -> createExtraSndFTDescrs db user fileId (map fileDescrText rfds)
-              case mapMaybe fileDescrURI rfds of
+              case mapMaybe (fileDescrURI fileInfo) rfds of
```
Comment (dpwiz) [inline src/Simplex/Chat.hs:3275]: Should be combined with #3876 to prevent redirect loops.

File: src/Simplex/Chat/Controller.hs [hunk 2]
```diff
@@ -1413,3 +1421,6 @@ $(JQ.deriveFromJSON defaultJSON ''ArchiveConfig)
 $(JQ.deriveFromJSON defaultJSON ''DBEncryptionConfig)
 
 $(JQ.deriveToJSON defaultJSON ''ComposedMessage)
+
+$(JQ.deriveFromJSON defaultJSON ''StandaloneFile)
+$(JQ.deriveToJSON defaultJSON ''StandaloneFile)
```
Comment (epoberezkin) [inline src/Simplex/Chat/Controller.hs:1426]:
```suggestion
$(JQ.deriveJSON defaultJSON ''StandaloneFile)
```

File: src/Simplex/Chat/Migrations/M20240307_file_info.hs [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+module Simplex.Chat.Migrations.M20240307_file_info where
+
+import Database.SQLite.Simple (Query)
+import Database.SQLite.Simple.QQ (sql)
+
+m20240307_file_info :: Query
+m20240307_file_info =
+  [sql|
+ALTER TABLE files ADD COLUMN file_info TEXT;
```
Comment (epoberezkin) [inline src/Simplex/Chat/Migrations/M20240307_file_info.hs:11]: we don't need it

File: src/Simplex/Chat/Styled.hs [hunk 4]
```diff
@@ -57,6 +58,10 @@ instance StyledFormat String where
   styled = Styled . sgr
   plain = Styled []
 
+instance StyledFormat LB.ByteString where
+  styled f = styled f . LB.unpack
+  plain = Styled [] . LB.unpack
```
Comment (epoberezkin) [inline src/Simplex/Chat/Styled.hs:63]: why not just use it in place... It's only one case
```suggestion
```

File: src/Simplex/Chat/View.hs [hunk 5]
```diff
@@ -218,6 +218,7 @@ responseToView hu@(currentRH, user_) ChatConfig {logLevel, showReactions, showRe
   CRSndFileError u (Just ci) _ -> ttyUser u $ uploadingFile "error" ci
   CRSndFileRcvCancelled u _ ft@SndFileTransfer {recipientDisplayName = c} ->
     ttyUser u [ttyContact c <> " cancelled receiving " <> sndFile ft]
+  CRStandaloneFileInfo info_ -> maybe ["no file information in URI"] (\j -> [plain $ J.encode j]) info_
```
Comment (epoberezkin) [inline src/Simplex/Chat/View.hs:221]:
```suggestion
  CRStandaloneFileInfo info_ -> maybe ["no file information in URI"] (\j -> [plain $ LB.unpack $ J.encode j]) info_
```

File: src/Simplex/Chat.hs [hunk 6]
```diff
@@ -2002,12 +2002,13 @@ processChatCommand' vr = \case
   StopRemoteCtrl -> withUser_ $ stopRemoteCtrl >> ok_
   ListRemoteCtrls -> withUser_ $ CRRemoteCtrlList <$> listRemoteCtrls
   DeleteRemoteCtrl rc -> withUser_ $ deleteRemoteCtrl rc >> ok_
-  APIUploadStandaloneFile userId file@CryptoFile {filePath} -> withUserId userId $ \user -> do
+  APIUploadStandaloneFile userId StandaloneFile {fileInfo, fileSource=file@CryptoFile {filePath}} -> withUserId userId $ \user -> do
```
Comment (epoberezkin) [inline src/Simplex/Chat.hs:2005]:
```suggestion
  APIUploadStandaloneFile userId StandaloneFile {fileInfo, fileSource = file@CryptoFile {filePath}} -> withUserId userId $ \user -> do
```

File: src/Simplex/Chat.hs [hunk 7]
```diff
@@ -2002,12 +2002,13 @@ processChatCommand' vr = \case
...
     fsFilePath <- toFSFilePath filePath
     fileSize <- liftIO $ CF.getFileContentsSize file {filePath = fsFilePath}
     when (fileSize > toInteger maxFileSizeHard) $ throwChatError $ CEFileSize filePath
-    (_, _, fileTransferMeta) <- xftpSndFileTransfer_ user file fileSize 1 Nothing
+    (_, _, fileTransferMeta) <- xftpSndFileTransfer_ user (decodeUtf8 . LB.toStrict . J.encode <$> fileInfo) file fileSize 1 Nothing
```
Comment (epoberezkin) [inline src/Simplex/Chat.hs:2009]:
```suggestion
    (_, _, fileTransferMeta) <- xftpSndFileTransfer_ user (safeDecodeUtf8 . LB.toStrict . J.encode <$> fileInfo) file fileSize 1 Nothing
```

**Pull request merged**